tests/actions/ReportTest.js

```diff
@@ -0,0 +1,73 @@
+import moment from 'moment';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '../../src/ONYXKEYS';
+import * as Pusher from '../../src/libs/Pusher/pusher';
+import PusherConnectionManager from '../../src/libs/PusherConnectionManager';
+import CONFIG from '../../src/CONFIG';
+import {subscribeToReportCommentEvents} from '../../src/libs/actions/Report';
+import waitForPromisesToResolve from '../utils/waitForPromisesToResolve';
+
+describe('actions/Report', () => {
+    it('should subscribe to the private-user-accountID channel and handle reportComment events', () => {
+        const REPORT_ID = 1;
+        const ACTION_ID = 1;
+        const REPORT_ACTION = {
+            actionName: 'ADDCOMMENT',
+            automatic: false,
+            created: 'Nov 6 2020 9:14am PST',
+            message: [{type: 'COMMENT', html: 'Testing a comment', text: 'Testing a comment'}],
+            person: [{type: 'TEXT', style: 'strong', text: 'Test User'}],
+            shouldShow: true,
+            timestamp: moment().unix(),
+            actorAccountID: 1,
+            sequenceNumber: ACTION_ID,
+            isAttachment: false,
+            loading: false,
+        };
+
+        // When using the Pusher mock the act of calling Pusher.isSubscribed will create a
+        // channel already in a subscribed state. These methods are normally used to prevent
+        // duplicated subscriptions, but we don't need them for this test.
+        Pusher.isSubscribed = jest.fn().mockReturnValue(false);
+        Pusher.isAlreadySubscribing = jest.fn().mockReturnValue(false);
+
+        // Connect to Pusher
+        PusherConnectionManager.init();
+        Pusher.init({
+            appKey: CONFIG.PUSHER.APP_KEY,
+            cluster: CONFIG.PUSHER.CLUSTER,
+            authEndpoint: `${CONFIG.EXPENSIFY.URL_API_ROOT}api?command=Push_Authenticate`,
+        });
+
+        let reportActions;
+        Onyx.connect({
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${REPORT_ID}`,
+            callback: val => reportActions = val,
+        });
+
+        // Set up Onyx with some test user data
+        Onyx.set(ONYXKEYS.SESSION, {accountID: 1, email: 'test@test.com'});
+        return waitForPromisesToResolve()
```

Comment:
I still haven't quite gotten the hang of these, so forgive me if this isn't how it's meant to be done... is the test that has been written any different than doing this?

```
return Onyx.set(ONYXKEYS.SESSION, {accountID: 1, email: 'test@test.com'})
```

My thinking is just that it's odd to see `waitForPromisesToResolve()` being used when the promise returned from `Onyx.set()` should resolve and work as well (and it's more obvious what is happening because the test uses the same promise chain).

Reply:
That makes sense! And I wouldn't mind updating it, but want to walk you through my thought process here to see if you have similar thoughts...

We could wait for `Onyx.set()` instead of doing `waitForPromisesToResolve()`, but I was thinking that for consistency we might want to default to `waitForPromisesToResolve()`. Even though unnecessary it might make it easier to understand how these tests are structured and deal with async code that doesn't necessarily have an "await-able" promise. Maybe this practice actually makes it harder to tell that's what `waitForPromisesToResolve()` is for?

Another thought I keep having is that if "everything async returned a `Promise`" (or even just the async "actions") then the test code might be more straight-forward and it would be more obvious to tell what exactly we are waiting for.

Reply:
I'll change this for now since I think I did have it this way at one point.

Reply:
OK, you brought up some really good points. What I'd like to avoid is creating a situation where people get too promise-happy and just return promises from everything. That essentially promotes a ton of promise chains, and that was what plagued the application from an early age. I had to do a lot of work to remove all those promise chains and I'd hate to go back down that path. So, now that I think about it, I think your original solution here is actually superior.